packages/commerce-sdk-react/src/auth/storage.ts

```diff
@@ -44,16 +44,25 @@ export class CookieStorage extends BaseStorage {
         super(options)
 
         if (typeof document === 'undefined') {
-            throw new Error('CookieStorage is not avaliable on the current environment.')
+            throw new Error('CookieStorage is not available on the current environment.')
         }
     }
     set(key: string, value: string, options?: Cookies.CookieAttributes) {
         const suffixedKey = this.getSuffixedKey(key)
+        const isInIframe = window.location !== window.parent.location
```

Comment:
```suggestion
        const isPreview = window.parent.location.hostname === 'runtime.commercecloud.com'
```

Reply:
If we go with this approach, we have to also check if parent is *.storefront.com or *.storefront-staging.com domain for testing/development purposes too. 🤔

Reply:
security wise, I think this approach is better than `window.location !== window.parent.location`, b/c that is "any site can iframe me and see my cookies",  I know we have the CSP guard at different places, but just in case customer mistakenly relaxes CSP.

Reply:
> If we go with this approach, we have to also check if parent is *.storefront.com or *.storefront-staging.com domain for testing/development purposes too. 🤔

🤔 yea, maybe not `*`, but a whitelist of internal hostnames?

Reply:
browser blocks cross origin frames from accessing values of the parent frame. So this isn't possible. Resolved